kube_aws_autoscaler/main.py

```diff
@@ -231,8 +237,42 @@ def slow_down_downscale(asg_sizes: dict, nodes_by_asg_zone: dict, scale_down_ste
     return asg_sizes
 
 
+def calculate_buffer_per_auto_scaling_group(autoscaling, nodes_by_asg_zone: dict,
+                                            buffer_percentage: dict, buffer_spare_nodes: int=0):
+    buffer_per_asg = {}
+
+    groups = []
+    for key in nodes_by_asg_zone:
+        asg_name, zone = key
+        groups.append(asg_name)
+
+    if len(groups) > 0:
+        response = autoscaling.describe_auto_scaling_groups(AutoScalingGroupNames=groups)
+        for asg in response['AutoScalingGroups']:
+            if 'Tags' in asg and asg['Tags']:
+                buffer_info = {}
+                for asg_tag in asg['Tags']:
+                    tag_key = asg_tag['Key']
+                    if tag_key in ASG_ALLOWED_TAGS:
+                        buffer_info[ASG_ALLOWED_TAGS[tag_key]] = int(asg_tag['Value'])
+
+                if buffer_info.keys():
```

Comment:
this if statement `if buffer_info.keys():` is not necessary as you check again in lines 261 & 264 as well.

Reply:
The reason for checking this is to make sure we only add buffer info for an ASG with valid tags. It'll avoid creating buffer_info for ASG with tags but none with `kube-aws-autoscaler:`